target/linux/ipq806x/image/generic.mk

```diff
@@ -547,6 +547,29 @@ define Device/ubnt_unifi-ac-hd
 endef
 TARGET_DEVICES += ubnt_unifi-ac-hd
 
+define Device/xiaomi_mi-router-hd
+	$(call Device/LegacyImage)
+	$(Device/dsa-migration)
+	DEVICE_VENDOR := Xiaomi
+	DEVICE_MODEL := Mi Router HD (R3D)
+	SOC := qcom-ipq8064
+	BLOCKSIZE := 128k
+	PAGESIZE := 2048
+	KERNEL_SIZE := 4096k
+	IMAGE_SIZE := 86016k
+	BOARD_NAME := mi-router-hd
+	SUPPORTED_DEVICES += xiaomi,r3d
+	UBINIZE_OPTS := -E 5
+	IMAGES += factory.bin
+	IMAGE/factory.bin := append-kernel | pad-to $$$$(KERNEL_SIZE) | \
+		append-ubi | pad-to $$$$(BLOCKSIZE) | check-size
+	DEVICE_PACKAGES := kmod-i2c-gpio kmod-thermal kmod-hwmon-lm75 \
+		kmod-hwmon-emc2305 hwmon-drivetemp kmod-usb-storage-uas \
```

Comment:
why usb storage kmod? I think that can be selected later by user? Do we do that also for other devices?

Reply:
> why usb storage kmod?
And without this module the USB refuses to work.

> Do we do that also for other devices?
Yes.
cisco_vedge1000, ubnt_edgerouter-e300, wevo_air-duo, dlink_dir-685, dlink_dns-313

Reply:
Generally UAS is not included by default, those that need it can always install it later or via AUC

Reply:
> i mean you need block-mount and additional package anyway...
> Generally UAS is not included by default, those that need it can always install it later or via AUC

But the same can be said about `hwmon-drivetemp` driver. 
This driver is pre-installed on many routers.

Also take a look at this place:
https://github.com/openwrt/openwrt/blob/683355d0a6d51cbeda258492bc16a9c6413b387e/target/linux/gemini/image/Makefile#L133-L142

Router Xiaomi R3D is also a NAS system.

Reply:
> Generally UAS is not included by default

Fresh commit: https://github.com/openwrt/openwrt/pull/16140
And this commit was checked by @robimarko 

https://github.com/openwrt/openwrt/blob/844701f1c045bf7d9bd06014a4b0cf9dd3d6a29d/target/linux/octeon/image/Makefile#L109